WordPress/openverse
# Fix waveform generation for short tracks

## Fixes



Fixes #4676 by @krysal 

## Description


This reduces the number of `pixels-per-second` required for tracks with 100 milliseconds of duration or less.

## Testing Instructions

I did the following to test this:

1. Insert this track in the catalog DB (it's the same report in Sentry). Run `j catalog/pgcli` and

```sql
INSERT INTO audio (
	identifier, 
	created_on, updated_on, 
	ingestion_type, 
	provider, source, 
	foreign_identifier, foreign_landing_url, 
	url, license, license_version, 
	creator, creator_url,
	duration,
	removed_from_source
) VALUES (
	'e0742ff3-f909-4e1d-9555-8e6399157452',
	NOW(),
	NOW(),
	'provider_api',
	'freesound',
	'freesound',
	'196927',
	'https://freesound.org/people/AlienXXX/sounds/196927',
	'https://cdn.freesound.org/previews/196/196927_97763-hq.mp3',
	'by',
	'4.0',
	'AlienXXX',
	'https://freesound.org/people/AlienXXX',
	45,
	false
);
```

2. Run the audio data refresh locally. `just a` & start the DAG from the Airflow UI.
3. Observe the peaks at the API endpoint for the track: http://localhost:50280/v1/audio/e0742ff3-f909-4e1d-9555-8e6399157452/waveform/

## Checklist


- [n/a] I added or updated documentation (if applicable).
- [n/a] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog PRs) or the media properties generator (`ov just catalog/generate-docs media-props` for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
